src/transceivercontroller/DefaultTransceiverController.ts

```diff
@@ -16,8 +16,43 @@ export default class DefaultTransceiverController implements TransceiverControll
 
   constructor(protected logger: Logger, protected browserBehavior: BrowserBehavior) {}
 
-  setEncodingParameters(_params: Map<string, RTCRtpEncodingParameters>): void {
-    return;
+  async setEncodingParameters(
+    encodingParamMap: Map<string, RTCRtpEncodingParameters>
+  ): Promise<void> {
+    if (!this._localCameraTransceiver || this._localCameraTransceiver.direction !== 'sendrecv') {
+      return;
+    }
+
+    const sender = this._localCameraTransceiver.sender;
+    if (!encodingParamMap || encodingParamMap.size === 0) {
+      return;
+    }
+    const newEncodingParams = Array.from(encodingParamMap.values());
+
+    const oldParam: RTCRtpSendParameters = sender.getParameters();
+    if (!oldParam.encodings) {
+      oldParam.encodings = newEncodingParams;
+    } else {
+      for (const existing of oldParam.encodings) {
+        for (const changed of newEncodingParams) {
+          if (existing.rid !== changed.rid) {
+            continue;
+          }
+          let key: keyof RTCRtpEncodingParameters;
+          for (key in changed) {
+            // These properties can't be changed.
+            if (key === 'rid' || key === 'codecPayloadType') {
+              continue;
+            }
+            if (changed.hasOwnProperty(key)) {
+              existing[key] = changed[key] as never;
+            }
```

Comment:
Is there a better way to model the value space of a keyof type? I haven't looked into this.

Reply:
So this is a workaround for the below TS error. Since there can be multiple types for existing[key], TS uses the intersection of all those types which is never to infer the type.
```
Type 'number | boolean | RTCDtxStatus' is not assignable to type 'never'.  
Type 'number' is not assignable to type 'never'
```

Alternatively, we can do:
- `(existing[key] as (number | boolean | RTCDtxStatus)) = changed[key];`
- or use ts-ignore.

Let me know if you can think of other ways :)

Reply:
```typescript
type paramValue = RTCRtpEncodingParameters[keyof RTCRtpEncodingParameters];
existing[key] as paramValue = changed[key];
```

should work.

Reply:
Nice. I will update the code.